src/main/java/core/basesyntax/Square.java

```diff
@@ -0,0 +1,32 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public class Square extends Figure {
+    private final double side;
+
+    {
+        setName("square");
+    }
+
+    public Square(Color color, double side) {
+        super(color);
+        this.side = side;
+    }
+
+    public Square() {
+        super(new ColorSupplier().getRandomColor());
+        this.side = new Random().nextInt(Figure.MAX_RANDOM_SIDE_VALUE);
```

Comment:
Check the message in the group chat. We can get rid of creating redundant instances of service classes if we pass them to the constructor.